### PR TITLE
Improve bound checking on scenery invention

### PR DESCRIPTION
**Pull request by tupaschoal**
Follow-up of #11067

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
I'm the first to defend ternary operators, but I agree with @Gymnasiast here. Once they start spanning several lines, it's time to reconsider their use.

**Comment by tupaschoal** [comment]
Done

File: src/openrct2/management/Research.cpp [hunk 1]
```diff
@@ -601,20 +601,29 @@ void ride_entry_set_invented(int32_t rideEntryIndex)
 
 bool scenery_is_invented(const ScenerySelection& sceneryItem)
 {
-    assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT);
-    return _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex];
+    Guard::Assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT, "Invalid Scenery Type");
```
Comment (Gymnasiast) [inline src/openrct2/management/Research.cpp]: Move the assert to the else as well and just have it assert `false`.

Comment (tupaschoal) [reply src/openrct2/management/Research.cpp]: Done.

File: src/openrct2/management/Research.cpp [hunk 2]
```diff
@@ -601,20 +601,29 @@ void ride_entry_set_invented(int32_t rideEntryIndex)
...
+        Guard::Assert(false, "Invalid Scenery Type");
+        return false;
+    }
 }
 
 void scenery_set_invented(const ScenerySelection& sceneryItem)
 {
-    assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT);
-    _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex] = true;
+    Guard::Assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT, "Invalid Scenery Type");
```
Comment (duncanspumpkin) [inline src/openrct2/management/Research.cpp:617]: I've always viewed Guard::Assert as an exit point of the program. I realise that isn't the case if you hit ignore but should we really be continuing. Wouldn't it be better to just `log_warning`

Comment (tupaschoal) [reply src/openrct2/management/Research.cpp:617]: Sure, I'd ditch the ignore anyway, if we could. In any case this does seem a bit out of the scope of this PR, no?

Comment (Gymnasiast) [reply src/openrct2/management/Research.cpp:617]: I prefer replacing Guard::Assert with `log_warning` as well.

Comment (tupaschoal) [reply src/openrct2/management/Research.cpp]: Ok, done

File: src/openrct2/management/Research.cpp [hunk 3]
```diff
@@ -601,20 +601,29 @@ void ride_entry_set_invented(int32_t rideEntryIndex)
...
+        log_warning("Invalid Scenery Type");
+        return false;
+    }
 }
 
 void scenery_set_invented(const ScenerySelection& sceneryItem)
 {
-    assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT);
-    _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex] = true;
+    log_warning("Invalid Scenery Type");
```
Comment (Gymnasiast) [inline src/openrct2/management/Research.cpp:617]: Er, what?! You're warning on every single call to `scenery_set_invented()`, including for valid data.

Comment (tupaschoal) [reply src/openrct2/management/Research.cpp:617]: Argh, my bad

Comment (tupaschoal) [reply src/openrct2/management/Research.cpp]: I've fixed it now.

**Comment by Gymnasiast** [comment]
Nice, that's three fewer places where it could potentially read and write out of bounds.

**Pull request merged**